modules/core/src/main/java/org/apache/ignite/internal/processors/cache/ExchangeContext.java

```diff
@@ -60,13 +61,19 @@
      * @param crd Coordinator flag.
      * @param fut Exchange future.
      */
-    public ExchangeContext(boolean crd, GridDhtPartitionsExchangeFuture fut) {
+    public ExchangeContext(boolean crd, GridDhtPartitionsExchangeFuture fut, IgniteLogger log) {
         int protocolVer = exchangeProtocolVersion(fut.firstEventCache().minimumNodeVersion());
 
+        boolean allNodesSupportsPmeFreeSwitch = allNodesSupports(fut.firstEventCache().allNodes(), PME_FREE_SWITCH);
+
+        if (!allNodesSupportsPmeFreeSwitch)
+            log.warning("Current topology does not support PME-free switch because " +
+                "not all nodes support IgniteFeatures." + PME_FREE_SWITCH + ".");
```

Comment:
'+ PME_FREE_SWITCH + "."'
What output we expect here?

Reply:
IgniteFeatures.PME_FREE_SWITCH. used one string